src/include/btmem.h

```diff
@@ -410,10 +409,24 @@ struct __wt_page {
 				uint32_t entries;
 				WT_REF	**index;
 			} * volatile __index;	/* Collated children */
+
+			/*
+			 * When splitting to deepen the tree, track the number
+			 * of entries in the newly created parent, and how many
+			 * subsequent splits follow the initial set of entries.
+			 * If future splits into the page are generally after
+			 * the initial set of items, perform future deepening
+			 * splits in this page to optimize for an append-style
+			 * workload.
+			 */
+			uint32_t deepen_split_append;
+			uint32_t deepen_split_last;
```

Comment:
This change didn't grow the page, so I don't think there's any need for this change.

I've been thinking today we need a bit from the WT_REF, set if the entry was inserted into the parent after the last split-deepen call -- that's all I think we need to detect real ranges.